### DISCUSSION TITLE
Using Kutt as internal tool for an organisation

### DISCUSSION THREAD

**Discussion by michael-gratton** [thedevs-network/kutt#532] [General]
Hi, I'm reviewing Kutt as a possible option for our public URL shortener - only staff would have access to the cutter, but redirects themselves would be public.

Would there be any interest in accepting patches implementing OAuth2 or SAML for single sign on? If so, any hints about a good place to start implementing it?

Is it possible (or planned) to make the client interface themeable? Also, would a simplified interface be achievable by removing interface elements present to support the kutt.it public service? Things like the report and github link, feature list and extension links. Our staff would contact the development team in case of bugs, and won't need to be convinced to use it. :)

**Comment by michael-gratton** [comment]
Thinking about the theming/simplifying a bit more, it might be taken care of if there was a way to configure a site-specific CSS file to be included in all page loads?

**Comment by poeti8** [comment]
Yes, we use a library called [passport](http://www.passportjs.org/docs/) for authentication. You can check out [passport.ts](https://github.com/thedevs-network/kutt/blob/develop/server/passport.ts) and [auth.ts](https://github.com/thedevs-network/kutt/blob/develop/server/handlers/auth.ts) to see how it's done and contribute if you want.

I'm not currently actively working on adding new features to Kutt, if you want to apply custom styles you should fork and host your own version of Kutt and apply custom changes.

**Comment by elmeligy** [comment]
hello
I have a similar setup
I installed kutt as an internal tool for my organization
I configures oauth2 proxy for SSO authentication for company users only, and it is working on the `/login` URL
I enabled (DISALLOW_ANONYMOUS_LINKS) so only authenticated users (company employees) can create links. 
but now I have to login twice , once by oauth , and then again for kutt
but I need the k8s ingress to auto login to kutt , how can I achieve this ?
